Jenkinsfile

```diff
@@ -192,21 +193,43 @@ pipeline {
                   returnStdout: true
                 )
                 KUBECONFIG = "${env.WORKSPACE}/${e2e_environment}/modules/k8s/secrets/admin.conf"
+                TESTPLAN = "MQ-1"
               }
               steps {
-                // FIXME(arne-rusek): move hcloud's config to top-level dir in TF scripts
-                sh """
-                  mkdir -p "${e2e_environment}/modules/k8s/secrets"
-                """
-                copyArtifacts(
-                    projectName: "${k8s_job.getProjectName()}",
-                    selector: specific("${k8s_job.getNumber()}"),
-                    filter: "${e2e_environment}/modules/k8s/secrets/admin.conf",
-                    target: "",
-                    fingerprintArtifacts: true
-                )
-                sh 'kubectl get nodes -o wide'
-                sh "nix-shell --run './scripts/e2e-test.sh --device /dev/sdb --tag \"${env.GIT_COMMIT_SHORT}\" --registry \"${env.REGISTRY}\"'"
+                script {
+                  // FIXME(arne-rusek): move hcloud's config to top-level dir in TF scripts
+                  sh """
+                    mkdir -p "${e2e_environment}/modules/k8s/secrets"
+                  """
+                  copyArtifacts(
+                      projectName: "${k8s_job.getProjectName()}",
+                      selector: specific("${k8s_job.getNumber()}"),
+                      filter: "${e2e_environment}/modules/k8s/secrets/admin.conf",
+                      target: "",
+                      fingerprintArtifacts: true
+                  )
+                  sh 'kubectl get nodes -o wide'
+                  sh 'mkdir -p ./e2e_junit'
+                  def pd = sh(script: 'pwd', returnStdout:true).trim()
+                  def junitdir = "${pd}/e2e_junit"
+                  sh "nix-shell --run './scripts/e2e-test.sh --device /dev/sdb --tag \"${env.GIT_COMMIT_SHORT}\" --registry \"${env.REGISTRY}\" --reportsdir \"${junitdir}\"'"
+                }
+              }
+              post {
+                always { // always send the junit results back to Xray and Jenkins
+                  junit 'e2e_junit/*.xml'
+                  script {
+                    step([
+                      $class: 'XrayImportBuilder',
+                      endpointName: '/junit',
+                      importFilePath: 'e2e_junit/*.xml',
+                      importToSameExecution: 'true',
+                      projectKey: "${projectkey}",
+                      testPlanKey: "${env.TESTPLAN}",
+                      serverInstance: "${env.JIRASERVERUUID}"])
+                    sh 'rm -f e2e_junit/*.xml'
```

Comment:
Please do not remove it. It will be automatically removed when new job is run in the same workspace and can be useful for debugging later. Or is it required?
```suggestion
```

Reply:
In my test pipeline they seem to hang around between runs. Maybe that is different in the Mayastor pipeline. I was just concerned that if we remove a test, old copies of the junit files will still be sent.
If you say they do get cleaned up, I will remove the line.

Reply:
I'll move the rm to before the test run, so we get to examine the files and they won't corrupt later runs.

Reply:
They **cannot** corrupt later runs as jenkins cleans up all untracked files before running the next job (`git clean -fdx` FYI we even tested it with @jkryl and it works) you do not need to clean up anything.

Reply:
Ok fair enough, I didn't know about the git clean action.